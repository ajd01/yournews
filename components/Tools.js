import React, { useContext } from 'react'
import { View, StyleSheet, Dimensions } from 'react-native'
import { Icon } from 'react-native-elements'
import { Context } from '../store'

export default function Tools ({ id }) {
  const { store, dispatch } = useContext(Context)

  const { rates, favorited } = store

  const rate = rates[id] || 0
  const favorite = favorited[id] || false

  const dispatchRate = (value) => {
    dispatch({
      type: 'setRate',
      value: value,
      id: id
    })
  }

  const dispatchFavorite = () => {
    dispatch({
      type: 'makeFavorite',
      id: id
    })
  }

  const starO = (value) => <Icon
    name='star-o'
    type='font-awesome'
    color='#f50'
    onClick={() => dispatchRate(value)}
  />
  const star = (value) => <Icon
    name='star'
    type='font-awesome'
    color='#f50'
    onClick={() => dispatchRate(value)}
  />

  const rateScore = () => {
    let elements = [false, false, false]
    elements = elements.map((e, i) => i < rate)
    return (
      <View style={styles.rate}>
        {elements.map((a, i) => a ? star(i + 1) : starO(i + 1))}
      </View>
    )
  }

  return (
    <View style={{ flexDirection: 'row' }}>
      <View style={styles.favorite}>
        {favorite
          ? <Icon
            name='heart'
            type='font-awesome'
            color='#f50'
            onClick={() => dispatchFavorite()}
            />
          : <Icon
            name='heart-o'
            type='font-awesome'
            color='#f50'
            onClick={() => dispatchFavorite()}
          />}

      </View>
      {rateScore()}
    </View>
  )
}

const styles = StyleSheet.create({
  favorite: {
    width: Dimensions.get('window').width / 4,
    flexDirection: 'row'
  },
  rate: {
    width: Dimensions.get('window').width / 4,
    flexDirection: 'row'
  }
})
