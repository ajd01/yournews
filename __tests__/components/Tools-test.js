import React from 'react'
import Tools from '../../components/Tools'
import renderer from 'react-test-renderer'
import {
  Context,
  InitialState
} from '../../store'

jest.mock('../../components/Icon', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    }
  }
})

const value = {
  dispatch: jest.fn(),
  store: {
    ...InitialState,
    data: [
      {
        author: 'Mockauthor',
        publishedAt: 'MockpublishedAt',
        title: 'Mocktitle',
        urlToImage: 'MockurlToImage',
        description: 'Mockdescription',
        content: 'Mockcontent',
        url: 'Mockurl'
      }
    ]
  }
}

describe('Tools componente', () => {
  it('renders initial screen', () => {
    const tree = renderer.create(
      <Context.Provider value={value}>
        <Tools />
      </Context.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
