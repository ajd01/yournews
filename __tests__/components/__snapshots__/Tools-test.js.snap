// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools componente renders initial screen 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": 187.5,
      }
    }
  >
    <div />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": 187.5,
      }
    }
  >
    <div />
    <div />
    <div />
  </View>
</View>
`;
